src/teleop/jetson/jetson_teleop.py

```diff
@@ -0,0 +1,149 @@
+        # turbo mode, multiply by 2 to make range [-1,1]
+        v = msg.linear.x
+        omega = msg.angular.z * 2
+
+        # Transform into L & R wheel angular velocities using diff drive kinematics
+        omega_l = (v - omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+        omega_r = (v + omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+
+        command = Chassis()
+        command.omega_l = omega_l
+        command.omega_r = omega_r
+
+        self.drive_vel_pub.publish(command)
+
+
+class ArmControl:
+    def __init__(self, xbox_mappings, ra_config):
+        self.xbox_mappings = xbox_mappings
+        self.ra_config = ra_config
+
+        # RA Joint Publishers
+        self.joint_a_pub = ros.Publisher("/ra/open_loop/joint_a", JointState, queue_size=100)
+        self.joint_b_pub = ros.Publisher("/ra/open_loop/joint_b", JointState, queue_size=100)
+        self.joint_c_pub = ros.Publisher("/ra/open_loop/joint_c", JointState, queue_size=100)
+        self.joint_d_pub = ros.Publisher("/ra/open_loop/joint_d", JointState, queue_size=100)
+        self.joint_e_pub = ros.Publisher("/ra/open_loop/joint_e", JointState, queue_size=100)
+        self.joint_f_pub = ros.Publisher("/ra/open_loop/joint_f", JointState, queue_size=100)
+
+        # Hand Publishers
+        self.finger_pub = ros.Publisher("/hand/open_loop/finger", JointState, queue_size=100)
+        self.grip_pub = ros.Publisher("/hand/open_loop/grip", JointState, queue_size=100)
+
+    def ra_control_callback(self, msg):
+
+        joint_a = JointState()
+        joint_a.velocity.append(
+            self.ra_config["joint_a"]["multiplier"]
+            * quadratic(deadzone(msg.axes[self.xbox_mappings["left_js_x"]], 0.15))
+        )
+        joint_b = JointState()
+        joint_b.velocity.append(
+            self.ra_config["joint_b"]["multiplier"]
+            * quadratic(-deadzone(msg.axes[self.xbox_mappings["left_js_y"]], 0.15))
+        )
+        joint_c = JointState()
+        joint_c.velocity.append(
+            self.ra_config["joint_c"]["multiplier"]
+            * quadratic(-deadzone(msg.axes[self.xbox_mappings["right_js_y"]], 0.15))
+        )
+        joint_d = JointState()
+        joint_d.velocity.append(
+            self.ra_config["joint_d"]["multiplier"]
+            * quadratic(deadzone(msg.axes[self.xbox_mappings["right_js_x"]], 0.15))
+        )
+        joint_e = JointState()
+        joint_e.velocity.append(
+            self.ra_config["joint_e"]["multiplier"]
+            * quadratic(
+                msg.buttons[self.xbox_mappings["right_trigger"]] - msg.buttons[self.xbox_mappings["left_trigger"]]
+            )
+        )
+        joint_f = JointState()
+        joint_f.velocity.append(
+            self.ra_config["joint_f"]["multiplier"]
+            * (msg.axes[self.xbox_mappings["right_bumper"]] - msg.axes[self.xbox_mappings["left_bumper"]])
+        )
+
+        self.joint_a_pub.publish(joint_a)
```

Comment:
For my own knowledge, can someone explain or link something about ROS message overhead? Why would we publish 6 messages with a single value instead of one message with 6 values? Also, what is the rationale for a list of velocities for each joint instead of a single float to represent the velocity?

Reply:
So by sending separate messages with a single value, there will probably be a little bit more overhead but it should be negligible. 

There are two reasons why I'd prefer each joint to have their own topic:
1. There already exists a joint state msg and I'd prefer not to create a custom msg when something already exists
2. If an application wanted to control only a single joint (either for debugging or something else), then grouping all of the joints together would make it so you need to also set the position/speed of the other joints too.

Reply:
But honestly I don't have too strong an argument for having each joint have their own topic. So if you want to, we can change back to having a single array if we think it's cleaner in code or if we are worried about code duplication

Reply:
This is maybe something to discuss when we all meet next, I think the idea behind separate topics is for assing certain joints in the sim software (rviz/gazebo) to a certain JointState topic.

Reply:
I'm not super opposed to having 6 different topics, both are fair points.